### PR TITLE
Java: Model Android Bundle and Intent extras methods

### PR DESCRIPTION
**Pull request by joefarebrother**

### CODE REVIEW

**Comment by smowton** [comment]
Could you run the test generator on these?

**Comment by smowton** [comment]
Also looks like there's a real test failure relating to existing modelling of the Activity.getIntent() / Bundle.getExtras() methods. Suggest the old-style models in FlowSources.qll and Intent.qll should be updated/removed at the same time to reduce confusion.

**Comment by smowton** [comment]
Nudge @joefarebrother

File: java/ql/lib/semmle/code/java/frameworks/android/Intent.qll [hunk 1]
```diff
@@ -52,3 +53,153 @@ class BundleGetterMethod extends Method, TaintPreservingCallable {
 
   override predicate returnsTaintFrom(int arg) { arg = -1 }
 }
+
+private class IntentBundleFlowSteps extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        //"namespace;type;subtypes;name;signature;ext;input;output;kind"
+        "android.os;BaseBundle;true;get;(String);;MapValue of Argument[-1];ReturnValue;value",
```
Comment (smowton) [inline java/ql/lib/semmle/code/java/frameworks/android/Intent.qll:62]: Why do you model considerably more put than get methods? Looks like the numeric types have models for the setters but not the getters?

Comment (joefarebrother) [reply java/ql/lib/semmle/code/java/frameworks/android/Intent.qll:62]: All of the put methods carry flow to the MapKey, so all are relavent, whereas MapValue flow from getters (and the second argument of putters) is ommitted for types that are generally considere to not carry taint (i.e. numeric types).

Comment (smowton) [reply java/ql/lib/semmle/code/java/frameworks/android/Intent.qll:62]: Aha, got it

File: java/ql/test/library-tests/frameworks/android/intent/Test.java [hunk 2]
```diff
@@ -0,0 +1,1499 @@
...
+// Test case generated by GenerateFlowTestCase.ql
+public class Test {
+
+	<T> T getElement(Iterable<T> it) { return it.iterator().next(); }
+	Object getIntent_extrasDefault(Object container) { return null; }
+	Object getMapKeyDefault(Object container) { return null; }
+	Object getMapValueDefault(Object container) { return null; }
+	Object newWithIntent_extrasDefault(Object element) { return null; }
+	Object newWithMapKeyDefault(Object element) { return null; }
+	Object newWithMapValueDefault(Object element) { return null; }
```
Comment (smowton) [inline java/ql/test/library-tests/frameworks/android/intent/Test.java:25]: Looks like these could mostly / all be filled in?

**Review by smowton — APPROVED** [review]
Excellent, thanks!

**Pull request merged**